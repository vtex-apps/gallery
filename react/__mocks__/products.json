[
  {
    "productId": "2",
    "productName": "Porsche 911",
    "link": "https://storecomponents.vtexcommercestable.com.br/porsche-911/p",
    "brand": "Porsche",
    "items": [
      {
        "name": "Porsche 911",
        "referenceId": [
          {
            "Value": "321",
            "__typename": "vtex_storegraphql_2_2_1_Reference"
          }
        ],
        "images": [
          {
            "imageUrl": "http://storecomponents.vteximg.com.br/arquivos/ids/155395/2017-porsche-911-carrera-s-coupe-angular-front.png?v=636583915191800000",
            "imageTag": "<img src=\"~/arquivos/ids/155395-#width#-#height#/2017-porsche-911-carrera-s-coupe-angular-front.png?v=636583915191800000\" width=\"#width#\" height=\"#height#\" alt=\"Porsche-911\" id=\"\" />",
            "__typename": "vtex_storegraphql_2_2_1_Image"
          }
        ],
        "sellers": [
          {
            "commertialOffer": {
              "Price": 220000,
              "ListPrice": 390000,
              "__typename": "vtex_storegraphql_2_2_1_Offer"
            },
            "__typename": "vtex_storegraphql_2_2_1_Seller"
          }
        ],
        "__typename": "vtex_storegraphql_2_2_1_SKU"
      }
    ],
    "__typename": "vtex_storegraphql_2_2_1_Product"
  },
  {
    "productId": "1",
    "productName": "Ninja 300 - Special Edition",
    "link": "https://storecomponents.vtexcommercestable.com.br/ninja-300/p",
    "brand": "Kawasaki",
    "items": [
      {
        "name": "Special Edition",
        "referenceId": [
          {
            "Value": "1235",
            "__typename": "vtex_storegraphql_2_2_1_Reference"
          }
        ],
        "images": [
          {
            "imageUrl": "http://storecomponents.vteximg.com.br/arquivos/ids/155394/17ninja300_wsb.jpg?v=636583737653170000",
            "imageTag": "<img src=\"~/arquivos/ids/155394-#width#-#height#/17ninja300_wsb.jpg?v=636583737653170000\" width=\"#width#\" height=\"#height#\" alt=\"17ninja300_wsb\" id=\"\" />",
            "__typename": "vtex_storegraphql_2_2_1_Image"
          }
        ],
        "sellers": [
          {
            "commertialOffer": {
              "Price": 24000,
              "ListPrice": 28000,
              "__typename": "vtex_storegraphql_2_2_1_Offer"
            },
            "__typename": "vtex_storegraphql_2_2_1_Seller"
          }
        ],
        "__typename": "vtex_storegraphql_2_2_1_SKU"
      }
    ],
    "__typename": "vtex_storegraphql_2_2_1_Product"
  },
  {
    "productId": "13",
    "productName": "Samsung S9 Plus",
    "link": "https://storecomponents.vtexcommercestable.com.br/samsung-s9/p",
    "brand": "Samsung",
    "items": [
      {
        "name": "S9 Plus",
        "referenceId": [
          {
            "Value": "87263191",
            "__typename": "vtex_storegraphql_2_2_1_Reference"
          }
        ],
        "images": [
          {
            "imageUrl": "http://storecomponents.vteximg.com.br/arquivos/ids/155408/samsung_galaxy_s9_plus_phone_template_2048x.jpg?v=636592276295770000",
            "imageTag": "<img src=\"~/arquivos/ids/155408-#width#-#height#/samsung_galaxy_s9_plus_phone_template_2048x.jpg?v=636592276295770000\" width=\"#width#\" height=\"#height#\" alt=\"front\" id=\"\" />",
            "__typename": "vtex_storegraphql_2_2_1_Image"
          },
          {
            "imageUrl": "http://storecomponents.vteximg.com.br/arquivos/ids/155409/download--1-.jpg?v=636592276669400000",
            "imageTag": "<img src=\"~/arquivos/ids/155409-#width#-#height#/download--1-.jpg?v=636592276669400000\" width=\"#width#\" height=\"#height#\" alt=\"back\" id=\"\" />",
            "__typename": "vtex_storegraphql_2_2_1_Image"
          }
        ],
        "sellers": [
          {
            "commertialOffer": {
              "Price": 3600,
              "ListPrice": 4000,
              "__typename": "vtex_storegraphql_2_2_1_Offer"
            },
            "__typename": "vtex_storegraphql_2_2_1_Seller"
          }
        ],
        "__typename": "vtex_storegraphql_2_2_1_SKU"
      }
    ],
    "__typename": "vtex_storegraphql_2_2_1_Product"
  },
  {
    "productId": "12",
    "productName": "Google Pixel - 2 XL",
    "link": "https://storecomponents.vtexcommercestable.com.br/google-pixel2/p",
    "brand": "Google",
    "items": [
      {
        "name": "2 XL",
        "referenceId": [
          {
            "Value": "6666",
            "__typename": "vtex_storegraphql_2_2_1_Reference"
          }
        ],
        "images": [
          {
            "imageUrl": "http://storecomponents.vteximg.com.br/arquivos/ids/155406/download.jpg?v=636592257640700000",
            "imageTag": "<img src=\"~/arquivos/ids/155406-#width#-#height#/download.jpg?v=636592257640700000\" width=\"#width#\" height=\"#height#\" alt=\"pixel-2\" id=\"\" />",
            "__typename": "vtex_storegraphql_2_2_1_Image"
          },
          {
            "imageUrl": "http://storecomponents.vteximg.com.br/arquivos/ids/155407/thumb_93290_default_big.jpg?v=636592257862070000",
            "imageTag": "<img src=\"~/arquivos/ids/155407-#width#-#height#/thumb_93290_default_big.jpg?v=636592257862070000\" width=\"#width#\" height=\"#height#\" alt=\"pixel-2xl\" id=\"\" />",
            "__typename": "vtex_storegraphql_2_2_1_Image"
          }
        ],
        "sellers": [
          {
            "commertialOffer": {
              "Price": 2200,
              "ListPrice": 2900,
              "__typename": "vtex_storegraphql_2_2_1_Offer"
            },
            "__typename": "vtex_storegraphql_2_2_1_Seller"
          }
        ],
        "__typename": "vtex_storegraphql_2_2_1_SKU"
      }
    ],
    "__typename": "vtex_storegraphql_2_2_1_Product"
  },
  {
    "productId": "8",
    "productName": "TV LG - 4K 3D",
    "link": "https://storecomponents.vtexcommercestable.com.br/tv-lg/p",
    "brand": "LG",
    "items": [
      {
        "name": "4K 3D",
        "referenceId": [
          {
            "Value": "6217361",
            "__typename": "vtex_storegraphql_2_2_1_Reference"
          }
        ],
        "images": [
          {
            "imageUrl": "http://storecomponents.vteximg.com.br/arquivos/ids/155401/tv-lg1.jpg?v=636592149449670000",
            "imageTag": "<img src=\"~/arquivos/ids/155401-#width#-#height#/tv-lg1.jpg?v=636592149449670000\" width=\"#width#\" height=\"#height#\" alt=\"lg-tv1\" id=\"\" />",
            "__typename": "vtex_storegraphql_2_2_1_Image"
          },
          {
            "imageUrl": "http://storecomponents.vteximg.com.br/arquivos/ids/155402/tv-lg2.jpg?v=636592149454870000",
            "imageTag": "<img src=\"~/arquivos/ids/155402-#width#-#height#/tv-lg2.jpg?v=636592149454870000\" width=\"#width#\" height=\"#height#\" alt=\"lg-tv2\" id=\"\" />",
            "__typename": "vtex_storegraphql_2_2_1_Image"
          }
        ],
        "sellers": [
          {
            "commertialOffer": {
              "Price": 2000,
              "ListPrice": 2000,
              "__typename": "vtex_storegraphql_2_2_1_Offer"
            },
            "__typename": "vtex_storegraphql_2_2_1_Seller"
          }
        ],
        "__typename": "vtex_storegraphql_2_2_1_SKU"
      }
    ],
    "__typename": "vtex_storegraphql_2_2_1_Product"
  },
  {
    "productId": "11",
    "productName": "Microsystem Sony - 100 RMS",
    "link": "https://storecomponents.vtexcommercestable.com.br/microsystem-sony/p",
    "brand": "Samsung",
    "items": [
      {
        "name": "100 RMS",
        "referenceId": [
          {
            "Value": "1293711",
            "__typename": "vtex_storegraphql_2_2_1_Reference"
          }
        ],
        "images": [
          {
            "imageUrl": "http://storecomponents.vteximg.com.br/arquivos/ids/155405/som-sony1.jpg?v=636592149480230000",
            "imageTag": "<img src=\"~/arquivos/ids/155405-#width#-#height#/som-sony1.jpg?v=636592149480230000\" width=\"#width#\" height=\"#height#\" alt=\"sony-som\" id=\"\" />",
            "__typename": "vtex_storegraphql_2_2_1_Image"
          }
        ],
        "sellers": [
          {
            "commertialOffer": {
              "Price": 1760,
              "ListPrice": 2000,
              "__typename": "vtex_storegraphql_2_2_1_Offer"
            },
            "__typename": "vtex_storegraphql_2_2_1_Seller"
          }
        ],
        "__typename": "vtex_storegraphql_2_2_1_SKU"
      }
    ],
    "__typename": "vtex_storegraphql_2_2_1_Product"
  },
  {
    "productId": "9",
    "productName": "Som LG - 100 RMS",
    "link": "https://storecomponents.vtexcommercestable.com.br/som-lg/p",
    "brand": "LG",
    "items": [
      {
        "name": "100 RMS",
        "referenceId": [
          {
            "Value": "9812983",
            "__typename": "vtex_storegraphql_2_2_1_Reference"
          }
        ],
        "images": [
          {
            "imageUrl": "http://storecomponents.vteximg.com.br/arquivos/ids/155400/som-lg1.jpg?v=636592149442230000",
            "imageTag": "<img src=\"~/arquivos/ids/155400-#width#-#height#/som-lg1.jpg?v=636592149442230000\" width=\"#width#\" height=\"#height#\" alt=\"som-samsung\" id=\"\" />",
            "__typename": "vtex_storegraphql_2_2_1_Image"
          }
        ],
        "sellers": [
          {
            "commertialOffer": {
              "Price": 1600,
              "ListPrice": 2000,
              "__typename": "vtex_storegraphql_2_2_1_Offer"
            },
            "__typename": "vtex_storegraphql_2_2_1_Seller"
          }
        ],
        "__typename": "vtex_storegraphql_2_2_1_SKU"
      }
    ],
    "__typename": "vtex_storegraphql_2_2_1_Product"
  },
  {
    "productId": "14",
    "productName": "Motorola Celular - Moto X4",
    "link": "https://storecomponents.vtexcommercestable.com.br/motorola-moto-x4/p",
    "brand": "Motorola",
    "items": [
      {
        "name": "Moto X4",
        "referenceId": [
          {
            "Value": "2863153",
            "__typename": "vtex_storegraphql_2_2_1_Reference"
          }
        ],
        "images": [
          {
            "imageUrl": "http://storecomponents.vteximg.com.br/arquivos/ids/155410/34324-5-smartphone-motorola-moto-x4-xt1900-32gb-tela-de-5-2-dual-chip-12-mp-8-mp-octa-core-e-3gb-ram-preto.jpg?v=636592277942030000",
            "imageTag": "<img src=\"~/arquivos/ids/155410-#width#-#height#/34324-5-smartphone-motorola-moto-x4-xt1900-32gb-tela-de-5-2-dual-chip-12-mp-8-mp-octa-core-e-3gb-ram-preto.jpg?v=636592277942030000\" width=\"#width#\" height=\"#height#\" alt=\"34324-5-smartphone-motorola-moto-x4-xt1900-32gb-tela-de-5-2-dual-chip-12-mp-8-mp-octa-core-e-3gb-ram-preto\" id=\"\" />",
            "__typename": "vtex_storegraphql_2_2_1_Image"
          },
          {
            "imageUrl": "http://storecomponents.vteximg.com.br/arquivos/ids/155411/detalhe-motorola-moto-x4-xt1900-azul-topazio.jpg?v=636592278294370000",
            "imageTag": "<img src=\"~/arquivos/ids/155411-#width#-#height#/detalhe-motorola-moto-x4-xt1900-azul-topazio.jpg?v=636592278294370000\" width=\"#width#\" height=\"#height#\" alt=\"detalhe-motorola-moto-x4-xt1900-azul-topazio\" id=\"\" />",
            "__typename": "vtex_storegraphql_2_2_1_Image"
          },
          {
            "imageUrl": "http://storecomponents.vteximg.com.br/arquivos/ids/155412/download.jpg?v=636592278711800000",
            "imageTag": "<img src=\"~/arquivos/ids/155412-#width#-#height#/download.jpg?v=636592278711800000\" width=\"#width#\" height=\"#height#\" alt=\"download\" id=\"\" />",
            "__typename": "vtex_storegraphql_2_2_1_Image"
          },
          {
            "imageUrl": "http://storecomponents.vteximg.com.br/arquivos/ids/155413/images.jpg?v=636592278890530000",
            "imageTag": "<img src=\"~/arquivos/ids/155413-#width#-#height#/images.jpg?v=636592278890530000\" width=\"#width#\" height=\"#height#\" alt=\"images\" id=\"\" />",
            "__typename": "vtex_storegraphql_2_2_1_Image"
          },
          {
            "imageUrl": "http://storecomponents.vteximg.com.br/arquivos/ids/155414/moto-x4-both-colors-2.jpg?v=636592279074070000",
            "imageTag": "<img src=\"~/arquivos/ids/155414-#width#-#height#/moto-x4-both-colors-2.jpg?v=636592279074070000\" width=\"#width#\" height=\"#height#\" alt=\"moto-x4-both-colors-2\" id=\"\" />",
            "__typename": "vtex_storegraphql_2_2_1_Image"
          },
          {
            "imageUrl": "http://storecomponents.vteximg.com.br/arquivos/ids/155415/smartphone-motorola-moto-x-x4-xt1900-32gb-12-0-mp-2-chips-android-7-1-nougat-3g-4g-wi-fi-photo196948602-12-20-3d.jpg?v=636592279477730000",
            "imageTag": "<img src=\"~/arquivos/ids/155415-#width#-#height#/smartphone-motorola-moto-x-x4-xt1900-32gb-12-0-mp-2-chips-android-7-1-nougat-3g-4g-wi-fi-photo196948602-12-20-3d.jpg?v=636592279477730000\" width=\"#width#\" height=\"#height#\" alt=\"smartphone-motorola-moto-x-x4-xt1900-32gb-12-0-mp-2-chips-android-7-1-nougat-3g-4g-wi-fi-photo196948602-12-20-3d\" id=\"\" />",
            "__typename": "vtex_storegraphql_2_2_1_Image"
          }
        ],
        "sellers": [
          {
            "commertialOffer": {
              "Price": 1540,
              "ListPrice": 1900,
              "__typename": "vtex_storegraphql_2_2_1_Offer"
            },
            "__typename": "vtex_storegraphql_2_2_1_Seller"
          }
        ],
        "__typename": "vtex_storegraphql_2_2_1_SKU"
      }
    ],
    "__typename": "vtex_storegraphql_2_2_1_Product"
  },
  {
    "productId": "10",
    "productName": "TV Sony - 4K 3D",
    "link": "https://storecomponents.vtexcommercestable.com.br/tv-sony/p",
    "brand": "Google",
    "items": [
      {
        "name": "4K 3D",
        "referenceId": [
          {
            "Value": "210399183",
            "__typename": "vtex_storegraphql_2_2_1_Reference"
          }
        ],
        "images": [
          {
            "imageUrl": "http://storecomponents.vteximg.com.br/arquivos/ids/155403/tv-sony.jpg?v=636592149473930000",
            "imageTag": "<img src=\"~/arquivos/ids/155403-#width#-#height#/tv-sony.jpg?v=636592149473930000\" width=\"#width#\" height=\"#height#\" alt=\"sony-tv1\" id=\"\" />",
            "__typename": "vtex_storegraphql_2_2_1_Image"
          },
          {
            "imageUrl": "http://storecomponents.vteximg.com.br/arquivos/ids/155404/tv-sony2.jpg?v=636592149477570000",
            "imageTag": "<img src=\"~/arquivos/ids/155404-#width#-#height#/tv-sony2.jpg?v=636592149477570000\" width=\"#width#\" height=\"#height#\" alt=\"sony-tv2\" id=\"\" />",
            "__typename": "vtex_storegraphql_2_2_1_Image"
          }
        ],
        "sellers": [
          {
            "commertialOffer": {
              "Price": 1330,
              "ListPrice": 1330,
              "__typename": "vtex_storegraphql_2_2_1_Offer"
            },
            "__typename": "vtex_storegraphql_2_2_1_Seller"
          }
        ],
        "__typename": "vtex_storegraphql_2_2_1_SKU"
      }
    ],
    "__typename": "vtex_storegraphql_2_2_1_Product"
  },
  {
    "productId": "7",
    "productName": "Som Bluetooth Samsung - 100 RMS",
    "link": "https://storecomponents.vtexcommercestable.com.br/som-samsung/p",
    "brand": "Samsung",
    "items": [
      {
        "name": "100 RMS",
        "referenceId": [
          {
            "Value": "9831831",
            "__typename": "vtex_storegraphql_2_2_1_Reference"
          }
        ],
        "images": [
          {
            "imageUrl": "http://storecomponents.vteximg.com.br/arquivos/ids/155397/som-samsung.jpg?v=636592103297600000",
            "imageTag": "<img src=\"~/arquivos/ids/155397-#width#-#height#/som-samsung.jpg?v=636592103297600000\" width=\"#width#\" height=\"#height#\" alt=\"som-samsung\" id=\"\" />",
            "__typename": "vtex_storegraphql_2_2_1_Image"
          }
        ],
        "sellers": [
          {
            "commertialOffer": {
              "Price": 936,
              "ListPrice": 1200,
              "__typename": "vtex_storegraphql_2_2_1_Offer"
            },
            "__typename": "vtex_storegraphql_2_2_1_Seller"
          }
        ],
        "__typename": "vtex_storegraphql_2_2_1_SKU"
      }
    ],
    "__typename": "vtex_storegraphql_2_2_1_Product"
  }
]