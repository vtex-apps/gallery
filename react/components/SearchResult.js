import React, { Component, Fragment } from 'react'
import { Spinner } from 'vtex.styleguide'
import { ExtensionPoint } from 'render'
import { FormattedMessage } from 'react-intl'

import LoadingOverlay from './LoadingOverlay'
import LayoutModeSwitcher from './LayoutModeSwitcher'
import { searchResultPropTypes } from '../constants/propTypes'
import OrderBy from './OrderBy'

/**
 * Search Result Component.
 */
export default class SearchResult extends Component {
  static propTypes = searchResultPropTypes

  state = {
    galleryLayoutMode: 'normal',
    showLoadingAsOverlay: false,
    // The definitions bellow are required because
    // on SSR the getDerivedStateFromProps isn't called
    products: this.props.products,
    recordsFiltered: this.props.recordsFiltered,
    brands: this.props.brands,
    map: this.props.map,
    params: this.props.params,
    priceRange: this.props.priceRange,
    priceRanges: this.props.priceRanges,
    rest: this.props.rest,
    specificationFilters: this.props.specificationFilters,
    tree: this.props.tree,
    hiddenFacets: this.props.hiddenFacets,
  }

  handleLayoutChange = (e, mode) => {
    e.preventDefault()

    this.setState({
      galleryLayoutMode: mode,
    })
  }

  static getDerivedStateFromProps(props) {
    // Do not use the props when the query is loading
    // so we can show the previous products when the
    // overlay is on the screen
    if (!props.loading) {
      const {
        products,
        recordsFiltered,
        brands,
        map,
        params,
        priceRange,
        priceRanges,
        rest,
        specificationFilters,
        tree,
        hiddenFacets,
      } = props

      return {
        products,
        recordsFiltered,
        brands,
        map,
        params,
        priceRange,
        priceRanges,
        rest,
        specificationFilters,
        tree,
        hiddenFacets,
      }
    }

    return null
  }

  componentDidMount() {
    if (!this.props.loading) {
      this.setState({
        showLoadingAsOverlay: true,
      })
    }
  }

  componentDidUpdate(prevProps) {
    if (prevProps.loading && !this.props.loading) {
      this.setState({
        showLoadingAsOverlay: true,
      })
    }
  }

  render() {
    const {
      children,
      breadcrumbsProps,
      getLinkProps,
      loading,
      fetchMoreLoading,
      summary,
      orderBy,
    } = this.props
    const {
      galleryLayoutMode,
      recordsFiltered,
      products,
      brands,
      map,
      params,
      priceRange,
      priceRanges,
      query,
      rest,
      specificationFilters,
      tree,
      hiddenFacets,
      showLoadingAsOverlay,
    } = this.state

    const term = params && params.term
      ? decodeURIComponent(params.term) : undefined

    if (!products.length && !loading) {
      return (
        <ExtensionPoint id="not-found" term={term} />
      )
    }

    const hideFacets = !map || !map.length

    const showLoading = loading && !fetchMoreLoading
    const showContentLoader = showLoading && !showLoadingAsOverlay

    return (
      <LoadingOverlay loading={showLoading && showLoadingAsOverlay}>
        <div className="vtex-search-result vtex-page-padding pv5 ph9-l ph7-m ph5-s">
          <div className="vtex-search-result__breadcrumb db-ns dn-s">
            <ExtensionPoint id="breadcrumb" {...breadcrumbsProps} />
          </div>
          <div className="vtex-search-result__total-products pb5 bn-ns bb-s b--muted-4 tc-s tl">
            <span className="vtex-search-result__term fw4 dn-ns db-s f4">
              {term}
            </span>
            <FormattedMessage
              id="search.total-products"
              values={{ recordsFiltered }}
            >
              {txt => <span className="ph4 c-muted-2">{txt}</span>}
            </FormattedMessage>
          </div>
          {!hideFacets && (
            <div className="vtex-search-result__filters">
              <ExtensionPoint
                id="filter-navigator"
                brands={brands}
                getLinkProps={getLinkProps}
                map={map}
                params={params}
                priceRange={priceRange}
                priceRanges={priceRanges}
                query={query}
                rest={rest}
                specificationFilters={specificationFilters}
                tree={tree}
                hiddenFacets={hiddenFacets}
                loading={loading && !fetchMoreLoading}
              />
            </div>
          )}
          <div className="vtex-search-result__border bg-muted-4 h-75 self-center" />
          <div className="vtex-search-result__order-by">
            <OrderBy
              orderBy={orderBy}
              getLinkProps={getLinkProps}
            />
          </div>
          <div className="vtex-search-result__gallery">
            <div className="dn-ns db-s bt b--muted-4">
              <LayoutModeSwitcher
                activeMode={galleryLayoutMode}
                onChange={this.handleLayoutChange}
              />
            </div>
            {showContentLoader ? (
              <div className="w-100 flex justify-center">
                <div className="w3 ma0">
                  <Spinner />
                </div>
              </div>
            ) : (
              <ExtensionPoint
                id="gallery"
                products={products}
                summary={summary}
                layoutMode={this.state.galleryLayoutMode}
              />
            )}
            {children}
          </div>
        </div>
      </LoadingOverlay>
    )
  }
}
