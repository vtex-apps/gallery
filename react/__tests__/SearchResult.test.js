import React from 'react'
import SearchResult from '../SearchResult'

import products from '../__mocks__/products.json'
import { mountWithIntl, loadTranslation } from 'enzyme-react-intl'

// import { getIntlContextInfo } from '../utils/intlHelpers'

describe('<SearchResult /> component', () => {
  function renderComponent(props = {}) {
    // const { context, childContextTypes, locale } = getIntlContextInfo()

    // loadTranslation(`../locales/${locale}.json`)

    const component = mountWithIntl(
      <SearchResult products={products} {...props} />,
      {
        context,
        // childContextTypes,
      },
    )

    return { component }
  }

  /* eslint-disable */
  it('should render the correct component', () => {
    const { component } = renderComponent()

    expect(component).toBeTruthy()
  })
})
