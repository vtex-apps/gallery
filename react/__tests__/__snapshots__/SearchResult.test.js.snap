// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResult /> component should match snapshot 1`] = `
<div>
  <div
    class="vtex-search-result w-100 pa3 dib"
  >
    <div
      class="w-100 w-30-m w-20-l fl pa3"
    >
      <div
        class="vtex-search-result__filter ph4 pt4 pb2 bb b--light-gray"
      >
        <div
          class="pointer mb4"
        >
          <div
            class="f4"
          >
            <span>
              Selected Filters
            </span>
            <span
              class="vtex-search-result__filter-icon fr"
            >
              <img
                src="[object Object]"
                width="20"
              />
            </span>
          </div>
        </div>
        <div
          class="ReactCollapse--collapse"
          style="overflow: hidden; height: 0px; overflow-y: auto; max-height: 200px;"
        >
          <div
            class="ReactCollapse--content"
          >
            <div
              class="w-90 db"
            >
              <a
                href="#"
              >
                <div
                  class="flex items-center mb3 relative"
                >
                  <div
                    class="h1 w1 relative ba bw1 br1 mr3 flex justify-center items-center b--silver bg-silver"
                  />
                  <div
                    class="absolute flex o-100 left-0"
                  >
                    <svg
                      class="vtex-icon__boxcheck"
                      height="16"
                      viewBox="0 0 14 14"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        fill="gray"
                      >
                        <path
                          d="M6.125 9.987L3.138 7l1.237-1.237 1.75 1.75 3.5-3.5 1.237 1.237z"
                        />
                      </g>
                    </svg>
                  </div>
                  <input
                    class="h1 w1 absolute o-0"
                    disabled=""
                    name="default-checkbox-group"
                    type="checkbox"
                    value=""
                  />
                  <label
                    class="near-black gray"
                  >
                    LANK
                  </label>
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="w-100 w-70-m w-80-l fl"
    >
      <div
        class="vtex-search-result__header flex flex-wrap justify-between"
      >
        <div
          class="vtex-search-result__header-search w-100 w-50-ns flex items-center justify-start pa3 ml7-l"
        >
          <span>
            Showing 1-0 from 0 records
          </span>
        </div>
        <div
          class="w-100 w-50-m w-20-l pa3"
        >
          <div
            class="vtex-dropdown"
          >
            <label>
              <div
                class="br2 bw1 relative bg-white ba b--light-gray hover-b--silver "
              >
                <div
                  class="vtex-dropdown__button bg-transparent bn w-100 pointer near-black f5 pv4 pl6 pr5 "
                >
                  <div
                    class="flex"
                  >
                    <div
                      class="vtex-dropdown__caption flex-auto tl truncate"
                    >
                      Sales
                    </div>
                    <div
                      class="vtex-dropdown__arrow flex-none flex items-center pl2"
                    >
                      <svg
                        height="18"
                        viewBox="0 0 24 24"
                        width="18"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          fill="#368DF7"
                        >
                          <path
                            d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z"
                          />
                        </g>
                      </svg>
                    </div>
                  </div>
                </div>
                <select
                  class="o-0 absolute top-0 left-0 w-100 bottom-0 pointer f5 "
                >
                  <option
                    value="OrderByTopSaleDESC"
                  >
                    Sales
                  </option>
                  <option
                    value="OrderByReleaseDateDESC"
                  >
                    Release Date
                  </option>
                  <option
                    value="OrderByBestDiscountDESC"
                  >
                    Discount
                  </option>
                  <option
                    value="OrderByPriceDESC"
                  >
                    Price: High to Low
                  </option>
                  <option
                    value="OrderByPriceASC"
                  >
                    Price: Low to High
                  </option>
                  <option
                    value="OrderByNameASC"
                  >
                    Name, ascending
                  </option>
                  <option
                    value="OrderByNameDESC"
                  >
                    Name, descending
                  </option>
                </select>
              </div>
            </label>
          </div>
        </div>
      </div>
      <div
        class="vtex-gallery pa3 w-100 flex flex-wrap justify-start"
      />
    </div>
  </div>
</div>
`;
